src/elfpy/simulators.py

```diff
@@ -36,23 +36,25 @@ def __init__(
         self,
```

Comment:
Adding the `agents` arg on init means the user has to pass two empty dictionaries to construct a Simulator that doesn't have any agents. It also seems like a likely point of confusion, we would have to be careful with documentation to explain the difference between the `init_agents` and `agents` args. I think we should be in agreement here as it sets expecations for the typical user workflow, so let's discuss. Here are two ideas for alternatives:

1. What do you think about passing only `init_agents` on Simulator initialization, and then creating a helper function `add_agents` that the user can call to add additional agents. This is more transparent than the previous workflow (which was `simulator.agents = get_example_agents(...)`), and addresses my above two complaints.

2. We could also remove the concept of `init_agents` entirely. Simulator is constructed without any agents passed.  The workflow would be then to always construct a Simulator without agents, and then manually add agents & execute trades as you see fit. Wrapper functions, like`run_simulation`, would include a check that the Simulator has agents. This way a user can be more purposeful about varying the number of agents in any given moment. This supports the notion that `collect_and_execute_trades` and `run_simulation` are wrapper functions, but not required for running simulations. One could make loops over days & blocks in an external script, and execute trades with varying sets of agents on any given day. The typical workflow (e.g. as set up in `example_main.py`) would then be to create the simulator, create `init_agent`, add `init_agent` to `simulator.agents`, execute trades, add more agents to `simulator.agents`, call `run_simulation`.

Reply:
I'm not a fan of (1). What do you think about adding an `initialize` method that does what the constructor does in this PR and otherwise adopting solution (2)? This would make sure that the agents list is empty to avoid any possible conflicts with the more manual workflow.

Reply:
yup I like that. If it's just adding agents then maybe `initialize_agents` as a better name?